internal/venice-common/src/test/java/com/linkedin/venice/writer/VeniceWriterUnitTest.java

```diff
@@ -533,4 +538,26 @@ public void testSendHeartbeat(boolean addLeaderCompleteHeader, LeaderCompleteSta
       }
     }
   }
+
+  // Write a unit test for the retry mechanism in VeniceWriter.close(true) method.
+  @Test(dataProvider = "True-and-False", dataProviderClass = DataProviderUtils.class, timeOut = 10 * Time.MS_PER_SECOND)
+  public void testVeniceWriterCloseRetry(boolean gracefulClose) throws ExecutionException, InterruptedException {
+    PubSubProducerAdapter mockedProducer = mock(PubSubProducerAdapter.class);
```

Comment:
We potentially should have some E2E integration test for this to test whether this will work for different scenarios or not.

Reply:
Sure, a new integration test is added.